### DISCUSSION TITLE
[FilePicker] Combine types and multiple in Android

### DISCUSSION THREAD

**Discussion by EranClientside** [capawesome-team/capacitor-plugins#69] [Q&A] [answered]
Hello everyone,

I saw that `types` is being ignored in Android when using `multiple: true`,
but I couldn't find any answer as to why it's implemented that way.

**Comment by robingenz** [comment, marked as answer]
That is a restriction in Android. The plugin cannot pass mime types if multiple is set to true. You need to implement your own logic for that (e.g. show a dialog if a file with invalid mime type is selected).

**Reply by EranClientside** [reply]
Thanks for the clarification.

**Comment by EranClientside** [comment]
I couldn't find any information about this restriction, so I did a test.
Removed the `multiple == false` from the condition and it worked.
```ts
const result = await FilePicker.pickFiles({
  multiple: true,
  types: [
    'application/msword',
	'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
	'application/vnd.ms-powerpoint',
	'application/vnd.openxmlformats-officedocument.presentationml.presentation',
	'application/vnd.ms-excel',
	'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
	'application/pdf',
	'text/plain',
	'text/csv',
  ],
});
```

Am I missing something? Is that something related to the target SDK version?

**Reply by robingenz** [reply]
I just tested it on my Pixel 6a with Android 13 and it does not work. As soon as I specify types, I can only select one file.

**Reply by eranshmil** [reply]
That's interesting. Work for me in Pixel 3a emulator and OnePlus 8T (Android 13 both).
I'll try to investigate that a little bit more.

**Reply by eranshmil** [reply]
I did some more tests using https://github.com/robingenz/capacitor-plugin-demo and it seems to work. (Pixel 6a emulator)

https://github.com/capawesome-team/capacitor-plugins/assets/95231509/b14afb56-ed9f-4ce3-9f73-6d3c5eeb4ec1

**Reply by robingenz** [reply]
Not for me. ^^


https://github.com/capawesome-team/capacitor-plugins/assets/13857929/f6547ef8-8f3f-4bd1-80b5-875c39664c04

**Reply by eranshmil** [reply]
Well, it seems like a different behavior than using single select.
Only double tap selects the file and closes the dialog.

https://github.com/capawesome-team/capacitor-plugins/assets/9721664/6578bb5e-b6f9-4413-9fd0-ef1c3459c27e

**Reply by robingenz** [reply]
Seems to me like we have different Android file pickers (UI is different). Probably because you are in the emulator.
